src/main/java/org/spongepowered/api/event/server/FullQueryEvent.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.event.server;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Called when the server is queried through the Query protocol, with more detailed information requested.
+ */
+public interface FullQueryEvent extends QueryEvent {
+
+    /**
+     * Gets the GameID to respond with.
+     *
+     * <p>By default, this is <code>MINECRAFT</code>.</p>
+     *
+     * @return The GameID to respond with.
+     */
+    String getGameID();
+
+    /**
+     * Sets the GameID to respond with.
+     *
+     * @param gameID The GameID to respond with.
+     */
+    void setGameID(String gameID);
+
+    /**
+     * Gets the version to respond with.
+     *
+     * <p>By default, this is the server's Minecraft version (e.g 1.8.1).</p>
+     *
+     * @return The version to respond with
+     */
+    String getVersion();
+
+    /**
+     * Sets the version to respond with.
+     *
+     * @param version The version to respond with
+     */
+    void setVersion(String version);
+
+    /**
+     * Gets the list of plugins to respond with.
+     *
+     * @return The list of plugins to respond with
+     */
+    String getPlugins();
```

Comment:
This is always blank in Vanilla, but CraftBukkit used it to display the CraftBukkit and Bukkit version, as well as the installed plugins. See [Dinnerbone's original post](https://dinnerbone.com/blog/2011/10/14/minecraft-19-has-rcon-and-query/)